modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/DiscountedCumulativeGainAt.java

```diff
@@ -178,19 +193,50 @@ private static double computeDCG(List<Integer> ratings) {
         PARSER.declareInt(DiscountedCumulativeGainAt::setUnknownDocRating, UNKNOWN_DOC_RATING_FIELD);
     }
 
+    @Override
+    public DiscountedCumulativeGainAt fromXContent(XContentParser parser, ParseFieldMatcher matcher) {
+        return DiscountedCumulativeGainAt.fromXContent(parser, new ParseFieldMatcherSupplier() {
+            @Override
+            public ParseFieldMatcher getParseFieldMatcher() {
+                return matcher;
+            }
```

Comment:
I think this can be simplified to DiscountedCumulativeGainAt.fromXContent(parser, () -> matcher).
Also I'm wondering why we need to duplicate the static method. I see this is necessary when implementing FromXContentBuilder<T>, but then again we were able do without this for the query builders for example. If `fromXContent()` should be a non-static thing, maybe we can get rid of the static implementation instead?